File: 1608-special-array-with-x-elements-greater-than-or-equal-x/1608-special-array-with-x-elements-greater-than-or-equal-x.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var specialArray = function(nums) {
    const n = nums.length;
    let count = Array(n+1).fill(0);

    for(let num of nums) {
        if(num >= n) count[n]++;
        else count[num]++;
    }

    let ans = 0;
    for(let i = n; i>0; i--) {
        ans += count[i];
        if(ans == i) return ans;
    }

    return -1;
};

File: 1608-special-array-with-x-elements-greater-than-or-equal-x/1608-special-array-with-x-elements-greater-than-or-equal-x.cpp
class Solution {
public:
    int specialArray(vector<int>& nums) {
        int n = nums.size();
        vector<int> count(n+1, 0);

        for(int num : nums) {
            if(num >= n) count[n]++;
            else count[num]++;
        }

        int ans = 0;
        for(int i = n; i>0; i--) {
            ans += count[i];
            if(i == ans) return ans;
        }

        return -1;
    }
};